### ISSUE TITLE
MongoDB with multiple connections ; hydrate problem

### ISSUE THREAD

**Issue by beesofts** [FriendsOfSymfony/FOSUserBundle#1992]
I'm using many connections to use data from many mongoDB databases. I followed the steps from symfony page to configure FOSUB. 
When I configure more than 1 connection for doctrine_mongodb, mapping configuration from FOS\UserBundle\Model\User don't seems to be loaded. I am able to persist documents, but reading won't work. I can read the fields I set in my own User class, but all fields from FOSUB class (email for example) is set to null.
Here is the config from my config.yml file:

```
doctrine_mongodb:
    connections:
        default:
            server: mongodb://%mongo_host%:%mongo_port%
            options:
                username: %mongo_user%
                password: %mongo_password%
        worker00:
            server: mongodb://%mongo_host%:%mongo_port%
            options:
                username: %mongo_user%
                password: %mongo_password%
    default_database: sociallymap
    document_managers:
        default:
            auto_mapping: true
        worker00:
            auto_mapping: true

fos_user:
    db_driver: mongodb
    firewall_name: main
    model_manager_name: default
    user_class: AppBundle\Document\User
```

This configuration works fine for my others documents.

If necessary I can provide a full sample project.

**Comment by stof** [comment]
We are registering the FOSUserBundle mapping only in the document manager you tell us to use (``default`` in this case). The ``worker00`` manager will not get the mapping (unless you register it yourselves)

**Comment by beesofts** [comment]
The user collection is under the default connection. When I set only one connection, all is going fine. When I set more than 1 connection, I'm not able to read the data that should be mapped by FOSUB whereas writing seems to work.

Here is a sample project that illustrate the problem I'm facing: https://github.com/beesofts/sample-fosub-mongo . That's only 2 commands to create and read data. If you comment the "local" mongo connection, all is going good. If there's more than 1 connection, the read command is able to display the $displayname field, but not the $email. I tried to set doctrine_mongodb.default_document_manager to default, or fos_user.model_manager_name to default, but this not works.

Am I missing something ?
